### PR TITLE
Solution for review

### PR DESCRIPTION
**Pull request by wladengine**

### CODE REVIEW

File: PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+﻿using Azure.Messaging.ServiceBus;
+
+namespace PhotoAndCopyShop.ScanWorker.Handlers;
+
+public class SendToServiceBusHandler : HandlerBase<FileInfo>
+{
+    private readonly string _connectionString;
+    private readonly string _queueName;
+    private const int MaxChunkSize = 250_000;
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs]: 250 k is too much, will lead to excessive LOH allocations

Comment (wladengine) [reply PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs]: I have improved memory management by direct file stream reading, now it should be less LOH-sensitive (at least, in client side)

File: PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    private readonly HashSet<string> _receivedMessages = new();
+
+    public ReceiverDaemon(string connectionString, string queueName, string targetFolder)
+    {
+        _connectionString = connectionString;
+        _queueName = queueName;
+        _targetFolder = targetFolder;
+    }
+
+    public async Task StartReceivingJobAsync()
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs:19]: This method looks excessively big and contains multiple "jump" operators like return and continue. Can you decompose it to smaller parts?

File: PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        // Create a Service Bus client using the connection string
+        await using ServiceBusClient client = new(_connectionString);
+        // Create a sender for the queue
+        ServiceBusReceiver receiver = client.CreateReceiver(_queueName);
+
+        while (true)
+        {
+            IAsyncEnumerable<ServiceBusReceivedMessage>? messages = receiver.ReceiveMessagesAsync();
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs]: This is never null

File: PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            }
+
+            await foreach (ServiceBusReceivedMessage message in messages)
+            {
+                if (_receivedMessages.Contains(message.MessageId))
+                {
+                    continue;
+                }
+                
+                _receivedMessages.Add(message.MessageId);
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs]: _receivedMessages Ids would be a bit more suitable

File: PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs (hunk 1 above)
Comment (ASemykinBrl) [reply PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs]: Anyway, it's better to keep it under 85 kb to ensure that, not critical

File: PhotoAndCopyShop.ScanWorker/FolderScannerDaemon.cs [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            string[] files = Directory.GetFiles(_folderPath);
+
+            foreach (string file in files)
+            {
+                var fileInfo = new FileInfo(file);
+
+                // check if the file was created within the last minute
+                if (DateTime.Now - fileInfo.CreationTime < TimeSpan.FromMinutes(1) && !_scannedFiles.Contains(fileInfo))
+                {
+                    await _handler?.HandleAsync(fileInfo);
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ScanWorker/FolderScannerDaemon.cs:32]: Suppose handle should never be null

File: PhotoAndCopyShop.ScanWorker/FolderScannerDaemon.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        _folderPath = folderPath;
+        _handler = handler;
+    }
+
+    public async Task ScanAsync()
+    {
+        Console.WriteLine($"Scanning folder {_folderPath} for new files...");
+
+        while (true)
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ScanWorker/FolderScannerDaemon.cs:21]: Can you implement some kind of circuit breaker to stop this process in a safe manner?

File: PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        // Create a sender for the queue
+        ServiceBusSender sender = client.CreateSender(_queueName);
+        try
+        {
+            // Create a message with the specified body
+            await using FileStream fs = new(fileInfo.FullName, FileMode.Open);
+            List<long> chunks = GetChunkedFileOffsets(fs.Length);
+            for (var i = 0; i < chunks.Count; i++)
+            {
+                var chunkData = new byte[MaxChunkSize];
```
Comment (ASemykinBrl) [inline PhotoAndCopyShop.ScanWorker/Handlers/SendToServiceBusHandler.cs]: If you carry this out of the loop, you will decrease allocations amount

File: PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs (hunk 2 above)
Comment (wladengine) [reply PhotoAndCopyShop.ReceiverService/ReceiverDaemon.cs:19]: extracted file-related write operations into SaveMessageData()

**Review by ASemykinBrl — APPROVED** [review]

**Pull request merged**